### DISCUSSION TITLE
Skip error files in file walk for registry scan(image)

### DISCUSSION THREAD

**Discussion by varanasikalyan** [aquasecurity/trivy#4856] [Bugs]
### Description

While doing a file walk with image scan(registry scan). I see we stop the scan with err when we couldn't open any file.

### Desired Behavior

I expected the scan should continue even when there is an err while opening the file.

![image](https://github.com/aquasecurity/trivy/assets/45425247/f5682cfc-c02d-47b4-8db7-f51a59218be7)

### Actual Behavior

scan failed with an err stating couldn't open file
![image](https://github.com/aquasecurity/trivy/assets/45425247/baad7132-044b-4347-b808-ccebc38070f2)

### Reproduction Steps

```bash
This issue happens when we perform image scan for an windows based image especially large windows images.
```

### Target

None

### Scanner

None

### Output Format

None

### Mode

None

### Debug Output

```bash
error in image inspect: analyze error: failed to analyze layer (sha256:358be10862ccbc3296389103d497dd7cd28b08c7931b4a545c88d7f7cd6): walk error: failed to process the file: failed to analyze file: failed to analyze Files/ProgramData/Microsoft/Windows Defender/Scans/mpcache-9164CD33A9A0DE10CD61559674A64B239F07C4.bin.5B: unable to open Files/ProgramData/Microsoft/Windows Defender/Scans/mpcache
9164CD33A9AFODE10CD61559674A64B239F07C4.bin.5B: failed to open: unable to read the file: unexpected EOF
```

### Operating System

windows

### Version

```bash
v0.41.1
```

### Checklist

- [ ] Run `trivy image --reset`
- [ ] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by knqyf263** [comment]
Is it reproducible?

**Reply by varanasikalyan** [reply]
I've not run the reproduction yet @knqyf263. Shall update you soon.

**Reply by varanasikalyan** [reply]
Hi @knqyf263 I've provided the details of the windows image for which the file walk is erroring out. As this is a private registry.
